File: 1350-remove-sub-folders-from-the-filesystem/remove-sub-folders-from-the-filesystem.java
class Solution {
    public List<String> removeSubfolders(String[] folder) {
        
        int i=0;
        Arrays.sort(folder);

        int n=folder.length;

        while(folder[i]=="/") i++;
        if(i>=n) return new ArrayList<>();

        List<String>result=new ArrayList<>();
        StringBuilder sb = new StringBuilder(folder[i]);
        result.add(folder[i]);

        for(int ind=i+1;ind<n;ind++){
            
            if(folder[ind].length() >= sb.length() && (folder[ind].substring(0,sb.length())).equals(sb.toString())){

                if(folder[ind].charAt(sb.length())=='/') continue;

                sb=new StringBuilder(folder[ind]);

                result.add(folder[ind]);
            }
            else{
                sb=new StringBuilder(folder[ind]);

                result.add(folder[ind]);
            }
        }
        return result;
    }
}

File: 1350-remove-sub-folders-from-the-filesystem/remove-sub-folders-from-the-filesystem.cpp
class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder) {

        sort(folder.begin(),folder.end());

        int n=folder.size();

        int i=0;

        while(i<n && folder[i]=="/")  i++;

        if(i>=n) return {};

        vector<string>result;

        result.push_back(folder[i]);
        string curr_string=folder[i];
        for(int ind=i+1;ind<n;ind++){ 
            
            if(folder[ind].substr(0,curr_string.length())==curr_string){
                int j = curr_string.length();
                if(folder[ind][j]=='/')
                    continue;
                curr_string=folder[ind];
                result.push_back(folder[ind]);
            }
            else{
                curr_string=folder[ind];
                result.push_back(folder[ind]);
            }
        }
        return result;
    }
};